substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/ELFObjectFile.java

```diff
@@ -1155,4 +1157,46 @@ public SymbolTable getSymbolTable() {
     protected int getMinimumFileSize() {
         return 0;
     }
+
+    @Override
+    public void installDebugInfo(DebugInfoProvider debugInfoProvider) {
+        DwarfSections dwarfSections = new DwarfSections(getMachine());
+        // we need an implementation for each section
+        DwarfSections.DwarfStrSectionImpl elfStrSectionImpl = dwarfSections.getStrSectionImpl();
+        DwarfSections.DwarfAbbrevSectionImpl elfAbbrevSectionImpl = dwarfSections.getAbbrevSectionImpl();
+        DwarfSections.DwarfFrameSectionImpl frameSectionImpl = dwarfSections.getFrameSectionImpl();
+        DwarfSections.DwarfInfoSectionImpl elfInfoSectionImpl = dwarfSections.getInfoSectionImpl();
+        DwarfSections.DwarfARangesSectionImpl elfARangesSectionImpl = dwarfSections.getARangesSectionImpl();
+        DwarfSections.DwarfLineSectionImpl elfLineSectionImpl = dwarfSections.getLineSectionImpl();
+        // now we can create the section elements with empty content
+        @SuppressWarnings("unused")
+        ELFSection strSection = (ELFSection) newUserDefinedSection(elfStrSectionImpl.getSectionName(), elfStrSectionImpl);
```

Comment:
No point in assigning to an unused local. Just use `newUserDefinedSection( ....`. I know the names of these methods are stupid.

Reply:
Yes, indeed. Removed.